src/NServiceBus.Core/SendContext.cs

```diff
@@ -0,0 +1,102 @@
+namespace NServiceBus
+{
+    using System;
+    using System.Collections.Generic;
+
+    /// <summary>
+    /// Allows the users to control how the send is performed
+    /// </summary>
+    public class SendContext
```

Comment:
These "context" classes are short lived instances, I wonder if we should make them structs to reduce heap allocations ? Just wondering